Gems/Atom/RHI/Vulkan/Code/Source/RHI/Scope.cpp

```diff
@@ -649,11 +649,32 @@ namespace AZ
         void Scope::OptimizeBarrier(const Barrier& unoptimizedBarrier, BarrierSlot slot)
         {
             uint32_t slotIndex = aznumeric_caster(slot);
-            AZStd::vector<Barrier>& barriers =
-                CanOptimizeBarrier(unoptimizedBarrier, slot) ? m_globalBarriers[slotIndex] : m_scopeBarriers[slotIndex];
+            bool canOptimize = CanOptimizeBarrier(unoptimizedBarrier, slot);
             auto filteredBarrier = unoptimizedBarrier;
             filteredBarrier.m_dstStageMask = RHI::FilterBits(unoptimizedBarrier.m_dstStageMask, m_deviceSupportedPipelineStageFlags);
-            filteredBarrier.m_srcStageMask = RHI::FilterBits(unoptimizedBarrier.m_srcStageMask, m_deviceSupportedPipelineStageFlags);            
+            filteredBarrier.m_srcStageMask = RHI::FilterBits(unoptimizedBarrier.m_srcStageMask, m_deviceSupportedPipelineStageFlags);
+
+            // a render pass can only automatically convert image layouts if a multi-aspect image has the same or compatible layouts for all
+            // aspects. This workaround covers the case for depth/stencil images where one aspect is in a transfer state:
+            if (canOptimize && filteredBarrier.m_type == VK_STRUCTURE_TYPE_IMAGE_MEMORY_BARRIER && UsesRenderpass() &&
+                IsRenderAttachmentUsage(filteredBarrier.m_attachment->GetUsage()) &&
+                (filteredBarrier.m_imageBarrier.subresourceRange.aspectMask == VK_IMAGE_ASPECT_DEPTH_BIT ||
+                 filteredBarrier.m_imageBarrier.subresourceRange.aspectMask == VK_IMAGE_ASPECT_STENCIL_BIT) &&
+                (filteredBarrier.m_imageBarrier.oldLayout == VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL ||
+                 filteredBarrier.m_imageBarrier.oldLayout == VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL))
```

Comment:
Do we maybe need a check if the image is actually depth+stencil, or is this also necessary for depth-only or stencil-only images?

Reply:
That would be nice, but at this point we can't check the image's aspects, since we only have the VkImage in the barrier and not the RHI::Image.